test/built-ins/String/prototype/matchAll/regexp-prototype-matchAll-v-u-flag.js

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2024 Tan Meng. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-regexp.prototype-@@matchall
+description: RegExp.prototype[@@matchAll] behavior with 'u'and 'v' flags.
+features: [Symbol.matchAll, regexp-v-flag, regexp-unicode-property-escapes]
+---*/
+
+const text = '𠮷a𠮷b𠮷';
+
+function doMatchAll(regex) {
+  const result = Array.from(RegExp.prototype[Symbol.matchAll].call(regex, text));
+  const matches = result.map(m => m[0]);
+  const indices = result.map(m => m.index);
+  return matches.concat(indices);
+}
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/𠮷/g), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "Basic matchAll with g flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/𠮷/gu), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "matchAll with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/𠮷/gv), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "matchAll with v flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/\p{Script=Han}/gu), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "Unicode property escapes with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/\p{Script=Han}/gv), ['𠮷', '𠮷', '𠮷', 0, 3, 6]),
+  undefined,
+  "Unicode property escapes with v flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/./gu), ['𠮷', 'a', '𠮷', 'b', '𠮷', 0, 2, 3, 5, 6]),
+  undefined,
+  "Dot with u flag"
+);
+
+assert.sameValue(
+  assert.compareArray(doMatchAll(/./gv), ['𠮷', 'a', '𠮷', 'b', '𠮷', 0, 2, 3, 5, 6]),
+  undefined,
+  "Dot with v flag"
+);
+
+assert.sameValue(
+  doMatchAll(/(?:)/gu).length,
+  6,
+  "Empty matches with u flag"
+);
+
+assert.sameValue(
+  doMatchAll(/(?:)/gv).length,
+  6,
+  "Empty matches with v flag"
+);
```

Comment:
These two tests seem to fail on my side on V8, JSC, SpiderMonkey, and Node. This is unexpected, since the second test was there already added in the previous PR #4213.

Reply:
Interesting, I think it's because I flat the array of [m[0], m.index] in `doMatchAll`.
I will fix this later.
Thanks for the feedback.